armory/utils/version.py

```diff
@@ -0,0 +1,134 @@
+'''
+ARMORY Versions use "Semantic Version" scheme where stable releases will have versions
+like `0.14.6`.  Armory uses the most recent git tag for versioning. For example if the
+most recent git tag is `v0.14.6`, then the version will be `0.14.6`.
+
+If you are a developer, the version will be constructed from the most recent tag plus a
+suffix of gHASH where HASH is the short hash of the most recent commit. For example,
+if the most recent git tag is v0.14.6 and the most recent commit hash is 1234567 then
+the version will be 0.14.6.g1234567. This scheme does differ from the scm strings
+which also have a commit count and date in them like 1.0.1.dev2+g0c5ffd9.d20220314181920
+which is a bit ungainly.
+'''
+
+import os
+import re
+import site
+import functools
+import setuptools_scm
+
+from pathlib import Path
+
+try:
+    from importlib import metadata
+except ImportError:
+    # Python <= 3.7
+    import importlib_metadata as metadata # noqa
+
+from armory.logs import log
+
+
+def to_docker_tag(version_str: str) -> str:
+    '''Convert version string to docker tag'''
+    return version_str.replace('+', '.')
+
+
+def get_metadata_version(package: str, version_str: str = '') -> str:
+    '''Retrieve the version from the package metadata'''
+    try:
+        return str(metadata.version(package))
+    except metadata.PackageNotFoundError:
+        log.warning(f"ERROR: Unable to find the specified package! Package {package} not installed.")
+    return version_str
+
+
+def get_tag_version(git_dir: Path = None) -> str:
+    '''Retrieve the version from the most recent git tag'''
+    project_paths = [Path(__file__).parent.parent, Path.cwd()]
```

Comment:
Remove the `Path.cwd()` - if you are getting the armory version but are in a non-armory git repo, this may attempt to get the armory version from that repo, unless I understand how the `scm_config` works.

Also, I think `Path(__file__).parent.parent` points to `project_root/armory`, not `project_root`.